tee-worker/core-primitives/types/src/lib.rs

```diff
@@ -132,6 +132,9 @@ pub enum TrustedOperationStatus {
 	/// TrustedOperation has been executed.
 	#[codec(index = 11)]
 	TopExecuted(Vec<u8>, bool),
+	/// TrustedOperation whose successor top has been executed, but the top itself not
+	#[codec(index = 12)]
+	SuccessorExecuted,
```

Comment:
I was very unconvinced about adding `TopExecuted` status to this enum because it is leaked to the client, this one looks even worse as it exposes unnecessary details but i assume there is no other easy way to fix it.

Reply:
Exactly - because of the fact that 3 related events could show up in arbitrary order(see comment in the issue), so we need a status to mark a connection hash as "can be removed later" in case the swap_hash operation shows up as the first event. 

Later when the desired status is reached the hash will get removed.  Otherwise the connection will be kept forever